refactor: To ensure telemetry is only enabled when both conditions are met.

Carpe-Wang opened the pull request:
# Description

About #339. Based on my understanding of this issue, to ensure telemetry is only enabled when both conditions are met, thus avoiding unnecessary or incorrect configuration leading to resource waste or incorrect data collection, I believe we need to add a check for `enableTelemetry` being true in the `if applicationInsightsID != ""{` and I want to change to `	if applicationInsightsID != "" && enableTelemetry {`. But I'm not sure if my understanding is correct.
## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

rbtr commented:
@Carpe-Wang this seems correct, can you test it and confirm the behavior is what we want?
Also need you to comment on #339 if you want it assigned to you 🙂

Carpe-Wang commented:
@microsoft-github-policy-service agree

Carpe-Wang commented:
> @Carpe-Wang this seems correct, can you test it and confirm the behavior is what we want? Also need you to comment on #339 if you want it assigned to you 🙂

hey @rbtr, I had comment on #339, and I noticed #339 had assigned to me. I make some new commits about test the unit test successfully.

nddq commented on init/retina/main_linux.go:
```diff
@@ -21,23 +22,25 @@ var (
+	}
+	l := zl.Named("init-retina").With(zap.String("version", version))
+	// Load configuration
+	cfg, err := config.GetConfig("/retina/config/config.yaml")
```
i don't think we should be hardcoding config file path

Carpe-Wang replied:
I use the `configPath := flag.String("config", "/default/path/to/config.yaml", "path to the config file")` to replace the hardcoding. Do you think it ok?

nddq replied:
yes, that is fine, but instead of `"/default/path/to/config.yaml"` you can use `/retina/config/config.yaml` since it is the default directory for config by default

Carpe-Wang replied:
Thank you for your help. I wasn't sure about the file path, so I changed it as you suggested.

nddq commented on init/retina/main_linux.go:
```diff
@@ -21,23 +24,27 @@ var (
+	if err != nil {
+		panic(err)
+	}
+	configPath := flag.String("config", "/default/path/to/config.yaml", "path to the config file")
```
```suggestion
	configPath := flag.String("config", "/retina/config/config.yaml", "path to the config file")
```

nddq reviewed (changes requested):
@Carpe-Wang can you sign all your commits please, thanks 🙂

Carpe-Wang commented:
> @Carpe-Wang can you sign all your commits please, thanks 🙂

Hey I have add GPG to commit.

Carpe-Wang commented:
> @Carpe-Wang can you sign all your commits please, thanks 🙂

@nddq I have signed all my commits. I hope they meet your expectations. Please check them.

nddq commented:
@Carpe-Wang seems like you missed a few commits

Carpe-Wang commented:
> @Carpe-Wang seems like you missed a few commits
@nddq yeah, those commits were unverified because the email in my local Git configuration didn't match the email for GPG. So, I deleted them and updated the email in my local Git configuration. However, the final code is consistent.

nddq commented:
> > @Carpe-Wang seems like you missed a few commits
> > @nddq yeah, those commits were unverified because the email in my local Git configuration didn't match the email for GPG. So, I deleted them and updated the email in my local Git configuration. However, the final code is consistent.

So what I meant is that we are looking for the `Verified` check across all commits, as you can see, some of them are missing that
![image](https://github.com/microsoft/retina/assets/28567936/29ad7042-758d-4c2a-a0ac-7591d5977577)
My suggestion is that you could squash all the previous commit into a single one, sign it and the force push it to this branch 🙂 Let me know if you have any issue

nddq commented:
@Carpe-Wang there are still unsigned commits, so we can't merge this yet 🥲

rbtr commented:
@Carpe-Wang as mentioned previously, **all commits** in a PR must be signed. As long as you have commits without a (Verified) badge [on this page](https://github.com/microsoft/retina/pull/393/commits), this will not be mergeable due to signing policy.
The easy way to resolve this is to squash the 21 commits currently on your branch down to a single one. Do this via interactive rebase, eg: `git rebase -i HEAD~21`, leave the first one as "pick", change all the rest to "squash", and adjust the commit message appropriately.

Carpe-Wang commented:
@nddq Sorry, I've been a bit busy these days and didn't have enough time to handle it. I just did a `reset --hard` now. If this doesn't meet your merge requirements, I'll make the changes again.

Carpe-Wang commented:
@rbtr thank you for your comment. I haven't used rebase before. I tried a few times without much success because when I pushed, there were many commits that weren't mine. So, I opted to use reset --hard directly. Thank you for your guidance.

nddq commented on init/retina/main_linux.go:
```diff
@@ -21,23 +24,31 @@ var (
+	}
+
+	// Define a command-line flag "config" with a default value and parse the flags, then create a logger with a name and version.
+	configPath := flag.String("config", "/retina/config/config.yaml", "path to the config file")
```
@Carpe-Wang looks like you need to pass in the path to the config via arguments to the init-container as well. Please refer to how we are currently doing it for the agent's main container
![image](https://github.com/microsoft/retina/assets/28567936/63bae1b6-ae9b-45dd-8b5f-dad6d2edf296)

Carpe-Wang replied:
<img width="714" alt="image" src="https://github.com/microsoft/retina/assets/78642589/f377b79e-46e9-4a0a-af65-6b0cd393fcd1">


---------------
I added the following section to the relevant file:
```
args:
  - --config=/retina/config/config.yaml
```
Does this meet your expectation?"

nddq replied:
that should work

Carpe-Wang replied:
okay, I had push the newest commit.

nddq replied:
oops i forgot to mention, you will also have to mount the path the config file like we are doing with the main container. Thanks for the prompt response!

Carpe-Wang replied:
@nddq I change the file according to your comment. Please review it.
